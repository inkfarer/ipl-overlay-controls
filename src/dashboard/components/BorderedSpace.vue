<template>
    <fieldset class="section-separator-wrapper">
        <legend>{{ label }}</legend>
        <slot />
    </fieldset>
</template>

<script lang="ts">
import { defineComponent } from 'vue';

export default defineComponent({
    name: 'BorderedSpace',

    props: {
        label: {
            type: String,
            required: true
        }
    }
});
</script>

<style lang="scss">
@import '../styles/colors';
@import '../styles/constants';

.section-separator-wrapper {
    border-color: #869FFF;
    border-radius: $border-radius-outer;
    padding: 8px;

    > legend {
        color: #869FFF;
    }
}
</style>
