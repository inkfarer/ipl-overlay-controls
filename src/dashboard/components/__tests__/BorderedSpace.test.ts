import { shallowMount } from '@vue/test-utils';
import BorderedSpace from '../BorderedSpace.vue';

describe('BorderedSpace', () => {
    it('matches snapshot', () => {
        const wrapper = shallowMount(BorderedSpace, {
            props: {
                label: 'Test Bordered Space'
            },
            slots: {
                default: '<div>test content</div>'
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });
});
