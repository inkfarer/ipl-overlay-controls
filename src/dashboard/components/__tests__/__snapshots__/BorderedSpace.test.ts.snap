// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BorderedSpace matches snapshot 1`] = `
"<fieldset class="section-separator-wrapper">
  <legend>Test Bordered Space</legend>
  <div>test content</div>
</fieldset>"
`;
